solr/core/src/java/org/apache/solr/core/ZkContainer.java

```diff
@@ -263,4 +275,27 @@ public void close() {
   public ExecutorService getCoreZkRegisterExecutorService() {
     return coreZkRegister;
   }
+
+  private void setGlobalUrlScheme(final ZkController zkCtrl) {
+    // Set the global urlScheme from cluster prop or if that is not set, look at the urlScheme sys prop
+    final String urlScheme = zkCtrl.getZkStateReader().getClusterProperty(ZkStateReader.URL_SCHEME, null);
+    if (StringUtils.isNotEmpty(urlScheme)) {
+      // track the urlScheme in a global so we can use it during ZK read / write operations for cluster state objects
+      GlobalStateVars.singleton().setUrlScheme(urlScheme);
+    } else {
+      final String urlSchemeFromSysProp = System.getProperty(URL_SCHEME, HTTP);
+      if (HTTPS.equals(urlSchemeFromSysProp)) {
+        // it's OK to set the cluster prop to https b/c we're making the cluster more secure, not less
+        // wouldn't do this for http though
+        log.warn("Cluster property 'urlScheme' not set but system property is set to 'https'. Updating the cluster property to match.");
```

Comment:
If we have an initially HTTP cluster, and then one node comes up with HTTPS, then that will bring down the rest of the cluster, right? I don't think we can do this rolling, however.

Reply:
I'll just remove this, not strictly necessary (was trying to save the user some work but could have side-effects)